src/graph/optimizer/rule/PushEFilterDownRule.cpp

```diff
@@ -123,17 +124,15 @@ std::string PushEFilterDownRule::toString() const {
         return nullptr;
       }
     } else {
-      ret = LogicalExpression::makeOr(pool);
-      std::vector<Expression *> operands;
-      operands.reserve(edges.size());
+      auto args = ArgumentList::make(pool);
       for (auto &edge : edges) {
         auto reEdgeExp = rewriteStarEdge(propertyExpr, spaceId, edge, schemaMng, pool);
         if (reEdgeExp == nullptr) {
           return nullptr;
         }
-        operands.emplace_back(reEdgeExp);
+        args->addArgument(reEdgeExp);
       }
-      static_cast<LogicalExpression *>(ret)->setOperands(std::move(operands));
+      ret = FunctionCallExpression::make(pool, "_any", args);
```

Comment:
Why not just rewrite it as `like.start_year==2010 or teammate.start_year==2010`?

Reply:
How about others?

Reply:
`like.start_year > xx' ?

Reply:
> `like.start_year > xx' ?

This filter could not be used in path pattern. I don't understand the purpose of the `_any` inner function.

Reply:
Storage don't support `*.prop`, so rewrite it.

Reply:
Agree with @czpmango comment.

Reply:
I think what @Shylock-Hg wants is a generic way. In the path pattern alone, however, rewriting with == suffices.